presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -1194,6 +1197,33 @@ public static long parseDuration(@SqlType("varchar(x)") Slice duration)
         }
     }
 
+    private static long timeAtTimeZone(ConnectorSession session, long timeWithTimeZone, TimeZoneKey timeZoneKey)
+    {
+        DateTimeZone sourceTimeZone = getDateTimeZone(unpackZoneKey(timeWithTimeZone));
+        DateTimeZone targetTimeZone = getDateTimeZone(timeZoneKey);
+        long millis = unpackMillisUtc(timeWithTimeZone);
+
+        // STEP 1. Calculate source UTC millis in session start
+        millis += valueToSessionTimeZoneOffsetDiff(session.getStartTime(), sourceTimeZone);
+
+        // STEP 2. Calculate target UTC millis in 1970
+        millis -= valueToSessionTimeZoneOffsetDiff(session.getStartTime(), targetTimeZone);
+
+        // STEP 3. Make sure that value + offset is in 0 - 23:59:59.999
+        long localMillis = millis + targetTimeZone.getOffset(0);
+        // Loops up to 2 times in total
+        while (localMillis > TimeUnit.DAYS.toMillis(1)) {
+            millis -= TimeUnit.DAYS.toMillis(1);
+            localMillis -= TimeUnit.DAYS.toMillis(1);
```

Comment:
We have `modulo24Hour` for this, instead of hand-rolled two loops.

Also, this potentially produces _negative_ `millis` value (or `millis > 24h` value), which is *probably* wrong representation.